controllers/resources/status.go

```diff
@@ -42,6 +43,7 @@ func UpdateStatusRemediationStarted(node *corev1.Node, nhc *remediationv1alpha1.
 			}
 			if !foundRem {
 				unhealthyNode.Remediations = append(unhealthyNode.Remediations, &remediation)
+				metrics.TrackNodeHealthCheckUnhealthyNodeOngoingRemediation(unhealthyNode.Name, remediation.Resource.Namespace, remediation.Resource.Kind)
```

Comment:
I would prefer to handle metrics where the CRs are actually created or deleted, alongside the event recording, not here where the status is updated. Would also have the advantage that you need to observe new CRs at one place only, and not two like here. WDYT?

Reply:
Good idea, I like it better like you suggested. We'd need to set the metric to 1 when the CR is created and then set it to 0 either when the CR is deleted or when the remediation times out.

I'll test it and make a new commit

Reply:
Do you think it would be good to differentiate deleted CR, from CR that timed out?

Reply:
> Do you think it would be good to differentiate deleted CR, from CR that timed out?

I don't think that needed. I think it's ok to see multiple remediations in that case in the metrics dashboard. You will see that they are created one after another. If you need details, you can look at the CRs or NHC logs.

Reply:
Maybe we can add a `true`/`false` label to the metric indicating whether the remediation timed out? Something like:

For currently-ongoing remediations:

`nodehealthcheck_ongoing_remediation{name="node-01", namespace="openshift-operators", remediation="SelfNodeRemediation", timedOut="false"} 1`

For timedout remediations:

`nodehealthcheck_ongoing_remediation{name="node-01", namespace="openshift-operators", remediation="SelfNodeRemediation", timedOut="true"} 0`

For successful remediations:

`nodehealthcheck_ongoing_remediation{name="node-01", namespace="openshift-operators", remediation="SelfNodeRemediation", timedOut="false"} 0`

Reply:
>  a customer might understand that the timeout set for a specific remediation is too low

They will already see that as soon as a 2nd remediation is created for the same node, not? Nothing else than a timeout on the former CR can cause that. 

Let's not overuse metrics and add new labels with low value, IMHO. Every label causes overhead in Prometheus.

Reply:
I think that for now we can probablu do without the metric being updated on timeout and only on CR deletion. Maybe something can be added in the future as per demand? 

I don't use `escalatingRemedies` as a user of NHC so I, personally, don't have a use case for it right now.

Reply:
So I tested it and it works.

But one possible downside to updating the metric when the `remediationCR` is created and the event is generated ([here](https://github.com/dana-team/node-healthcheck-operator/blob/feature/ongoing-remediation-metric/controllers/nodehealthcheck_controller.go#L460)) is that if the controller pod is recreated before the remediation is complete, then metric will not be updated since the CR already exists, and there will just be no data available for the metric until the `remediationCR` is deleted.

If the update is done in the status then the metric will be (more or less) continuous

Reply:
>  if the controller pod is recreated before the remediation is complete, then metric will not be updated since the CR already exists

Why would the metric need an update in this case? 🤔

Reply:
Because the remediation is still ongoing, no? But it's a `remediationCR` not created by the current instance of NHC

Reply:
The metric was set by the old NMO instance and is persistet in Prometheus, not? 🤔 
Or maybe I need to update myself how exactly Prometheus works 🙈 

However, in the status code the metric would also not be updated, because the status is persistet for sure 🙂 If we need to set metrics on pod restart, the call needs to be at another place either way.

Reply:
ah, wait, Prometheus is polling for metrics regularly, that's why, correct?

Reply:
That's right, yes. I tested it locally as well

Reply:
ack
I would put the ObeserveCreated call here then: https://github.com/medik8s/node-healthcheck-operator/blob/main/controllers/nodehealthcheck_controller.go#L476